src/main/java/com/odysseusinc/athena/service/impl/VocabularyServiceImpl.java

```diff
@@ -206,11 +217,68 @@ public DownloadBundle saveDownloadItems(DownloadBundle bundle, List<Integer> idV
     }
 
     @Override
-    public List<DownloadBundleDTO> getDownloadHistory(Long userId) {
+    public List<DownloadBundleDTO> getDownloadHistory(AthenaUser user) {
 
         Sort sort = new Sort(Sort.Direction.DESC, "created");
-        List<DownloadBundle> history = downloadBundleRepository.findByUserId(userId, sort);
-        return converterUtils.convertList(history, DownloadBundleDTO.class);
+        List<DownloadBundle> history = downloadBundleRepository.findByUserId(user.getId(), sort);
+        Set<Long> bundleIds = history.stream()
+                .map(b -> b.getId())
+                .collect(Collectors.toSet());
+
+        List<DownloadShare> shares = downloadShareRepository.findByDownloadShareIdUserEmail(user.getEmail());
+        List<DownloadBundleDTO> sharedDTOs = new ArrayList<>();
+        // add shared bundles to list of available downloads
+        if (!shares.isEmpty()) {
+            for(DownloadShare share: shares) {
```

Comment:
There are lot of queries and conversions to get list of shares.

Reply:
refactoring made